src/libcore/iter/adapters/mod.rs

```diff
@@ -1752,6 +1752,95 @@ where
     }
 }
 
+/// An iterator that only accepts elements while `predicate` returns `Some(_)`.
+///
+/// This `struct` is created by the [`map_while`] method on [`Iterator`]. See its
+/// documentation for more.
+///
+/// [`map_while`]: trait.Iterator.html#method.map_while
+/// [`Iterator`]: trait.Iterator.html
+#[must_use = "iterators are lazy and do nothing unless consumed"]
+#[unstable(feature = "iter_map_while", reason = "recently added", issue = "68537")]
+#[derive(Clone)]
+pub struct MapWhile<I, P> {
+    iter: I,
+    finished: bool,
```

Comment:
The whole point of the `MapWhile` is that it stops on first `None` just like `TakeWhile` stops on first `false`

Reply:
It stops by having `.next()` return `None`. The `finished` field isn't needed to do that. After then it doesn't matter what `.next()` returns unless the iterator implements `FusedIterator` which `MapWhile` doesn't.

Reply:
I first thought it was to improve the upper bound of `size_hint`, but it also doesn't matter what that method returns once the iterator has been exhausted. So I suppose you're right. We probably can't remove it from `TakeWhile` though, since it already implements `FusedIterator`.

Reply:
It can be a bit confusing, so if we do remove `finished` flag, then I think we need to mention it in the docs